scheduler/app/libs/tasks/treasury_ops.py

```diff
@@ -32,7 +35,14 @@ def add_statistics(
         if prev == 0 or current == 0:
             returns.append(0)
         else:
-            returns.append(log(current / prev))
+            _return: float = None
+            try:
+                _return = log(current / prev)
+            except ValueError as e:
```

Comment:
This fix does not render line 35 obsolete as this is not the only case where there is a domain error with the `ln` function.

As per the (bootleg) `print_exception` log statement on line 43:
```bash
[2022-06-07 15:02:18,709: WARNING/ForkPoolWorker-2] ValueError: math domain error
[2022-06-07 15:02:18,709: WARNING/ForkPoolWorker-2] curr: 291397.0261113758, prev: -503930.95555524435
```
Which makes sense because `-x` is indeed not included in the domain of `ln`: `(0, inf)` or all real, positive integers.

Actually thinking about it more, I should just add that to the clause on line 35 😅. But really with these `print_exception` "log" statements I just wanted to show the conditions for error so that we can discuss them (I 100% agree with you earlier suggestion that we should utilize a logging lib. Maybe we can piggy back off the logging lib celery utilizes).

Reply:
> Also in case we end up in the except clause, it seems you are appending None to returns. Shouldn't we rather append 0? (I'm not sure)
This will be corrected by modifying the clause on line 35 👍.